src/cloud/aws/custom/paws.pm

```diff
@@ -839,6 +840,68 @@ sub directconnect_describe_virtual_interfaces {
     return $results;
 }
 
+sub cloudtrail_events {
+    my ($self, %options) = @_;
+
+    my $events_results = [];
+    eval {
+        my $ct;
+        if (length $self->{option_results}->{endpoint}) {
+            $ct = $self->{paws}->service('CloudTrail', region => $self->{option_results}->{region} , endpoint => $self->{option_results}->{endpoint});
+        } else {
+            $ct = $self->{paws}->service('CloudTrail', region => $self->{option_results}->{region});
+        }
+        my %ct_options = ();
+        if (defined($options{delta})) {
+            $ct_options{EndTime} = time();
+            $ct_options{StartTime} = $ct_options{EndTime} - ($options{delta} * 60);
+        }
+
+        while (my $list_events = $ct->LookupEvents(%ct_options)) {
+            foreach (@{$list_events->{Events}}) {
+                my $event = JSON::XS->new->utf8->decode($_->{CloudTrailEvent});
```

Comment:
i think you don't need to json decode.

Why you don't use the same method ? https://docs.aws.amazon.com/cli/latest/reference/cloudtrail/get-trail-status.html

Reply:
it doesn't work without the json decode because (it is a json string as a value in the json returned)

Reply:
indeed